cloudflare_dns: fix KeyError

lucasbasquerotto opened the pull request:
##### SUMMARY
Fixes https://github.com/ansible-collections/community.general/issues/236

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
cloudflare_dns

##### ADDITIONAL INFORMATION
Fix errors of the type `KeyError: 'success'` to show a more meaningful message, like:

`API user does not have permission; Status: 401; Method: DELETE: Call: /zones/a8222778e37a0276468013d671b39e15/dns_records/b8051d66e565436cd6f24fbd34621f52; Error details: You cannot use this API for domains with a .cf, .ga, .gq, .ml, or .tk TLD (top-level domain). To configure the DNS settings for this domain, use the Cloudflare Dashboard.`

felixfontein commented:
Please add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to) (`bugfixes:`).

Andersson007 commented on plugins/modules/net_tools/cloudflare_dns.py:
```diff
@@ -496,7 +496,11 @@ def _cf_simple_api_call(self, api_call, method='GET', payload=None):
 
-        if not result['success']:
+        if not result.get('success'):
+            if not 'success' in result:
```
```
        if not result.get('success'):
            if not 'success' in result:
```
the whole construction looks a bit overengineering:)
@lucasbasquerotto could we make it simpler?

lucasbasquerotto commented:
I've made a new commit.

@Akasurde I've corrected the `not in` clause.

@Andersson007 I've separated the 2 nested ifs to make it simpler.

@felixfontein Changelog fragment added.

I've also removed the trailing whitespaces in the empty line as reported by the bot.
